libraries/chain/proposal_evaluator.cpp

```diff
@@ -164,14 +159,30 @@ void_result proposal_update_evaluator::do_apply(const proposal_update_operation&
          p.available_key_approvals.insert(id);
       for( const auto& id : o.key_approvals_to_remove )
          p.available_key_approvals.erase(id);
+      try
+      {
+         // first check authorization
+         execution_authorized = p.is_authorized_to_execute(d, p.fail_reason);
+         // now check to see if there is a review period
+         if (p.review_period_time)
+         {
+            execution_authorized = false;
+            if(p.fail_reason.empty())
+               p.fail_reason = "Awaiting expiration of review period";
+         }
+      }
+      catch (fc::exception& ex)
+      {
+         p.fail_reason = ex.to_string(fc::log_level(fc::log_level::all));
+      }
    });
 
    // If the proposal has a review period, don't bother attempting to authorize/execute it.
    // Proposals with a review period may never be executed except at their expiration.
    if( _proposal->review_period_time )
       return void_result();
 
-   if( _proposal->is_authorized_to_execute(d) )
+   if( execution_authorized )
    {
       // All required approvals are satisfied. Execute!
       _executed_proposal = true;
```

Comment:
Actually what I specifically wanted for this issue is to update the reason in `catch` below.